project-base/storefront/store/slices/createPageLoadingStateSlice.ts

```diff
@@ -14,13 +14,15 @@ const CUSTOM_PAGE_TYPES = [
 export type PageType = FriendlyPagesTypesKey | (typeof CUSTOM_PAGE_TYPES)[number];
 
 export type PageLoadingStateSlice = {
+    hadClientSideNavigation: boolean;
```

Comment:
Why we do have past tense here? 🤔 I know I will be little annoying (with `is` again :D) but what about presence tense like `isClientSideNavigation`?

Reply:
The question I want an answer to is "had the client side navigation run already?". I need to know this in order to differentiate between when defer should be used and where it should not. This name makes sense to me. I can rename it to `didClientSideNavigationRun` or  `hasClientSideNavigationRun`, if that makes more sense to you.

Reply:
I looked at the code again to understand better but realized I am actually more confused. So maybe we can think about different naming than all of those. Because now I am asking myself why in the begging of the navigation I am setting value which is saying that navigation has ended...? 🤔

![image](https://github.com/shopsys/shopsys/assets/11465991/7cbeb885-23fe-47ea-8a46-871853c517e6)

Reply:
The entire point of this variable is to divide the lifetime of the application into 2 parts:

1. before the first client side navigation has started
2. after the first client side navigation has started

This would make the natural name candidate to be "hasClientSideNavigationStarted" or "hadClientSideNavigationStarted", which would theoretically make sense, but I just like it better without the "Started" suffix, which is unnecessary in my opinion. I do not know about any other option for the names. The main goal is to have a boolean pointer which tells the developer which of those two parts of application's lifetime he is currently in.